src/tiled/propertybrowser.cpp

```diff
@@ -1618,7 +1618,7 @@ void PropertyBrowser::updateProperties()
         if (auto visibleProperty = mIdToProperty[VisibleProperty])
             visibleProperty->setValue(mapObject->isVisible());
         mIdToProperty[XProperty]->setValue(mapObject->x());
-        mIdToProperty[YProperty]->setValue(InvertYAxisHelper().getPixelY(mapObject->y()));
+        mIdToProperty[YProperty]->setValue(InvertYAxisHelper(MapDocumentActionHandler::instance()->mapDocument()->map()).pixelY(mapObject->y()));
```

Comment:
In `applyMapObjectValueTo`, you used `mMapDocument`. You should do the same here to make sure the right map is used.